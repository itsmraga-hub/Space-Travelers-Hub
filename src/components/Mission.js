const Mission = (props) => {
  const mission = props;
  const {
    name, desc, id, join, isReserved,
  } = mission;

  return (
    <li>
      <h2>{ name }</h2>
      <p>{ desc }</p>
      <a rel="noreferrer" href="##">
        {
        isReserved ? 'Active Member' : 'Not A Member'
      }
      </a>
      <button type="button" id={id} onClick={join}>
        {
          isReserved ? 'Leave Mission' : 'Join Mission'
        }
      </button>
    </li>
  );
};

export default Mission;
