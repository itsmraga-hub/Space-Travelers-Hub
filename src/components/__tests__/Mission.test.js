import renderer from 'react-test-renderer';

import Mission from '../Mission';

const mission = {
  name: 'SpaceX',
  desc: 'Lorem ipsum text generate',
  id: '56DFRE',
  join: () => true,
  reserved: false,
};

const mission2 = {
  name: 'SpaceX',
  desc: 'Lorem ipsum text generate',
  id: '56DFRE',
  join: () => true,
  reserved: true,
};

it('<Mission /> renders properly as expected', () => {
  const {
    name, desc, id, join, reserved,
  } = mission;
  const tree = renderer.create(<Mission
    name={name}
    id={id}
    desc={desc}
    join={join}
    reserved={reserved}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('<Mission /> renders properly as expected', () => {
  const {
    name, desc, id, join, reserved,
  } = mission2;
  const tree = renderer.create(<Mission
    name={name}
    id={id}
    desc={desc}
    join={join}
    reserved={reserved}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
