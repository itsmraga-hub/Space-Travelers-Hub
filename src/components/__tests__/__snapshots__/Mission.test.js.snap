// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mission /> renders properly as expected 1`] = `
<li
  className="Mission"
>
  <p
    className="Title"
  >
    SpaceX
  </p>
  <p
    className="desc"
  >
    Lorem ipsum text generate
  </p>
  <div
    data-id={true}
  >
    <a
      className=" badge"
      href="##"
      rel="noreferrer"
    >
      Not A Member
    </a>
  </div>
  <button
    className=" btn"
    id="56DFRE"
    onClick={[Function]}
    type="button"
  >
    Join Mission
  </button>
</li>
`;

exports[`<Mission /> renders properly as expected 2`] = `
<li
  className="Mission"
>
  <p
    className="Title"
  >
    SpaceX
  </p>
  <p
    className="desc"
  >
    Lorem ipsum text generate
  </p>
  <div
    data-id={true}
  >
    <a
      className=" badge"
      href="##"
      rel="noreferrer"
    >
      Not A Member
    </a>
  </div>
  <button
    className=" btn"
    id="56DFRE"
    onClick={[Function]}
    type="button"
  >
    Join Mission
  </button>
</li>
`;
